lib/collection/src/common/sha_256.rs

```diff
@@ -25,6 +41,52 @@ pub async fn hash_file(file_path: &Path) -> io::Result<String> {
     Ok(format!("{hash:x}"))
 }
 
+/// Compute sha256 hash for the given file using the `sha256sum` binary
+///
+/// May fail if the binary does not exist on the system.
+#[cfg(all(debug_assertions, unix))]
+async fn hash_file_bin(file_path: &Path) -> io::Result<String> {
+    use tokio::process::Command;
+
+    let output = match Command::new("sha256sum")
+        .arg("-b")
+        .arg(file_path.as_os_str())
+        .output()
+        .await
+    {
+        // Binary must have success exit code
+        Ok(command) if !command.status.success() => {
+            return Err(io::Error::new(
+                io::ErrorKind::Other,
+                format!(
+                    "Failed to compute checksum with sha256sum: {}",
+                    String::from_utf8(command.stderr).unwrap(),
+                ),
+            ));
+        }
+        // Binary must have been called successfully
+        Err(err) => {
+            return Err(io::Error::new(
+                io::ErrorKind::Other,
+                format!("Failed to compute checksum with sha256sum: {err}"),
+            ));
+        }
+        Ok(command) => command,
```

Comment:
I'd use `Ok(command) if command.status.success() => command` at the top.

```suggestion
        Ok(command) if command.status.success() => command,
        Ok(command) => {
            return Err(io::Error::new(
                io::ErrorKind::Other,
                format!(
                    "Failed to compute checksum with sha256sum: {}",
                    String::from_utf8(command.stderr).unwrap(),
                ),
            ));
        }
        Err(err) => {
            return Err(io::Error::new(
                io::ErrorKind::Other,
                format!("Failed to compute checksum with sha256sum: {err}"),
            ));
        }
```